src/ops/control_flow.jl

```diff
@@ -315,20 +331,28 @@ Example using shape_invariants:
                 end
                 extend_graph(def_graph, g_def.node[to_delete])
                 deleteat!(g_def.node, unique(to_delete))
-                for collection in [:Variables, :TrainableVariables]
-                    for var in get_collection(g, collection)
-                        name = get_def(var.var_node).name
-                        as_default(def_graph) do
-                            Variable(name)
-                        end
+                @assert get_collection(:TrainableVariables) ⊆  get_collection(:Variables)
```

Comment:
This line throws an error on travis:
`TypeError: non-boolean (TensorFlow.Tensor{Bool}) used in boolean context`

Have we got an overload that breaks this, or am I making a silly mistake?

Reply:
Mmm not sure. I think I'll just merge for now and then try to fix it.